### PR TITLE
deps: Bump `com_github_nghttp2_nghttp2` -> 1.49.0

### LINKED ISSUE #22809: Newer release available `com_github_nghttp2_nghttp2`: v1.49.0 (current: v1.48.0)
Package Name: com_github_nghttp2_nghttp2@1.48.0
Current Version: v1.48.0@2022-06-24
Available Version: v1.49.0@2022-08-22
Upstream releases: https://github.com/nghttp2/nghttp2/releases

### PR DESCRIPTION
**Pull request by phlax**
Fixes #22809

Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -453,12 +453,12 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "Nghttp2",
         project_desc = "Implementation of HTTP/2 and its header compression algorithm HPACK in Cimplementation of HTTP/2 and its header compression algorithm HPACK in C",
```
Comment (moderation) [inline bazel/repository_locations.bzl:454]: Can we use this PR to fix this long outstanding typo?
`       project_desc = "Implementation of HTTP/2 and its header compression algorithm HPACK in C",`

**Review by moderation — CHANGES_REQUESTED** [review]

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**